### ISSUE TITLE
Create standard `wellcomeml.viz` output to visualise predictions

### ISSUE THREAD

**Issue by aCampello** [wellcometrust/WellcomeML#222]
We could potentially have a class (wrapping [BertViz](https://github.com/jessevig/bertviz) or another tool) to visualise the predictions of the bert classifier or others?

**Comment by piyumalanthony** [comment]
We can add other BERT variants and the visualizations by wrapping up visualization with BertViz.  Is there a possibility of adding other classifiers here as well like XLM and Roberta?

**Comment by aCampello** [comment]
We are definitely open to adding new classifiers, in particular ones that perform well on scientific datasets (we are moving towards specialised models, hence why we added 'scibert').

But I believe adding other classifiers should be opened as a different issue, as I'd like us to focus on having a nice visualisation util for this one.

**Comment by piyumalanthony** [comment]
Yes, Focusing only on the BERT could be better. My suggestion is to add a visualization strategy to the contextual embedding of BERT and attention patterns of BERT initially. Furthermore, visualizations for training and validations could've added in the second phase. Any suggestions for further use cases?

**Comment by aCampello** [comment]
Also bringing to the your attention (see what I did there) this framework by AllenNLP on explaining predictions (https://allennlp.org/interpret)

**Comment by piyumalanthony** [comment]
Thanks for the information. I will take that into the consideration.

**Comment by nsorros** [comment]
Late to the part but it would be great to get something either simple as BertViz or more complicated that allows us to see what the attentions layers are attending in our Bert models (we are currently mainly using Bert and SciBert)

**Comment by piyumalanthony** [comment]
Thanks for the comment @nsorros. The initial plan is to develop the visualization to the level of BertViz and in the second phase, the visualization could be further developed considering the tasks performed by the model.

**Comment by nsorros** [comment]
> Thanks for the comment @nsorros. The initial plan is to develop the visualization to the level of BertViz and in the second phase, the visualization could be further developed considering the tasks performed by the model.

Sounds like a good plan 👍

**Comment by federikovi** [comment]
So the scope of this project would be implementing `BertViz` on the existing `wellcomeml` code or doing it ex novo? For the visualisation of the embeddings I only knew tensorflow [embedding projector](https://projector.tensorflow.org/), BertViz is really nice!

**Comment by piyumalanthony** [comment]
BertViz only provides visualizations for the attention mechanism and under this project, the initial plan is to consider scibert and add visualization for attention and tasks based considerations.
